lms/djangoapps/email_marketing/tasks.py

```diff
@@ -294,6 +296,72 @@ def _get_course_content(course_url, sailthru_client, email_config):
     return response
 
 
+def _get_or_create_user_lists(sailthru_client, site_domain=None):
+    """
+    Get the user list name from cache if exists else create one and return the name
+    :param sailthru_client:
+    :param user_lists:
+    :return: list name if exists or created else return None
+    """
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
+
+    if site_domain and "edx.org" not in site_domain:
+        list_name = site_domain.replace(".", "_") + "_user_list"
+    else:
+        list_name = "All edX Users"
+
+    cached = cache.get(email_config.CACHE_KEY)
+    if cached:
+        for user_list in cached['lists']:
+            if user_list.get('name') == list_name:
+                return list_name
+
+    sailthru_response = _get_or_create_list(sailthru_client, list_name)
+    if sailthru_response:
+        cache.set(email_config.CACHE_KEY, sailthru_response)
+        return list_name
+
+    return
+
+
+def _get_or_create_list(sailthru_client, list_name):
+    """
+    Get list from sailthru and return if list_name exists else create a new one
+    and return list data for all lists.
+    """
+    sailthru_get_response = sailthru_client.api_get("list", {})
+    if not sailthru_get_response.is_ok():
+        error = sailthru_get_response.get_error()
+        log.info("Error attempting to read list record from Sailthru: %s", error.get_message())
+        return
+
+    for user_list in sailthru_get_response.json['lists']:
+        if user_list.get('name') == list_name:
+            return sailthru_get_response.json
+
+    list_params = {'list': list_name, 'primary': 0, 'public_name': list_name}
```

Comment:
#nitpick, add a comment to say from this point, we are going to create the user list